pumpkin-protocol/src/lib.rs

```diff
@@ -68,6 +68,18 @@ pub enum SoundCategory {
     Voice,
 }
 
+#[derive(Serialize)]
+pub struct IDOrSoundEvent {
+    pub id: VarInt,
+    pub sound_event: Option<SoundEvent>,
+}
```

Comment:
Im not sure what this is Struct is. Looking at Vanilla code the Packets do just need the SoundEvent

Reply:
This Struct is an implementation from the wiki.vg page. However, the wiki.vg page seems to be wrong in multiple locations for these packets, they did update it recently but it still wrong. Maybe this part is wrong as well. I will give it a look.

Reply:
Just looked at the vanilla code and it seems like it only sends SoundEvent but it must be adding an additional zero hex while encoding the object. Without the extra 0x00 sent, the client does not accept the additional string bytes. The vanilla codebase must only use the strings and thus the additional id parameter is not necessary for them to implement. We can do the same and just remove this struct in favor of a encoded 0x00.